pkg/api/vlabs/validate.go

```diff
@@ -1041,20 +1046,35 @@ func validatePasswordComplexity(name string, password string) (out bool) {
 }
 
 // Validate validates the KubernetesConfig
-func (k *KubernetesConfig) Validate(k8sVersion string, hasWindows bool) error {
+func (k *KubernetesConfig) Validate(k8sVersion string, hasWindows, ipv6DualStackEnabled bool) error {
 	// number of minimum retries allowed for kubelet to post node status
 	const minKubeletRetries = 4
 
+	// ipv6 dual stack feature is currently only supported with kubenet
+	if ipv6DualStackEnabled && k.NetworkPlugin != "kubenet" {
+		return errors.Errorf("OrchestratorProfile.KubernetesConfig.NetworkPlugin '%s' is invalid. IPv6 dual stack supported only with kubenet.", k.NetworkPlugin)
+	}
+
 	if k.ClusterSubnet != "" {
-		_, subnet, err := net.ParseCIDR(k.ClusterSubnet)
-		if err != nil {
-			return errors.Errorf("OrchestratorProfile.KubernetesConfig.ClusterSubnet '%s' is an invalid subnet", k.ClusterSubnet)
+		clusterSubnets := strings.Split(k.ClusterSubnet, ",")
+		if !ipv6DualStackEnabled && len(clusterSubnets) > 1 {
```

Comment:
I think these validation errors should complain about `OrchestratorProfile.KubernetesConfig.ClusterSubnet` instead of `OrchestratorProfile.KubernetesConfig.NetworkPlugin`